tests/ui/methods/probe-for-diagnostic-doesnt-do-extra-work.rs

```diff
@@ -0,0 +1,157 @@
+#![allow(non_snake_case)]
```

Comment:
Excellent that you have a test case! Please add a brief comment explaining what this is testing and pointing to this PR.